openfisca_core/entities/group_entity.py

```diff
@@ -23,7 +27,15 @@ class GroupEntity(Entity):
 
     """  # noqa RST301
 
-    def __init__(self, key, plural, label, doc, roles, containing_entities=()):
+    def __init__(
+        self,
+        key: str,
+        plural: str,
+        label: str,
+        doc: str,
+        roles: Sequence[HasKey],
```

Comment:
Is there any reason why the roles should be ordered? I don't see one and would recommend to use `Set` or `Iterable` instead.

Reply:
I would not change that ionce it might be used by package dependent on core (and even core) when filling population if the roles are indexed by an integer.

Reply:
After spending a couple of hours on this... good part of the code depends on roles and sub-roles being ordered.

Reply:
I prefer not to touch that for now, not before the aforesaid parts of the code are properly tested and documented.

Reply:
BTW @MattiSG at least in Python, `iterable` is a generic protocol including `list` (ordered, mutable), `tuple` (ordered, immutable), `dict` (not-ordered, mutable), and `set` (not-ordered, sort of mutable).

@benjello Why? You mean, for example, I have a GroupEntity of 4 people. You'll use role/subrole's index to determine each person's order within the group? (I actually had strange results when I tried to remove indexation).